### PR TITLE
feat(experience): implement the consent page

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Implement the experience consent page.

This PR includes the following updates:
1. previously for all the applications, we automatically call the consent API and consent all the missing scope directly without user interaction.   With the third-party application implemented, we updated the consent page to display the detail info of all the requested permissions that user needs to grant,  including OIDC user profile permissions, resource permissions and organizations. 

2. Update the landing page `BrandingHeader` component to support the display of the third-party app logo if provided. 
3. Update the consent page to display all the requested permission details and organizations. 
4. When the user click on the `Authorize` button call the consent API 

image

## Testing

test locally
will add integration test later

## Checklist

- [ ] ~`.changeset`~
- [ ] ~unit tests~
- [ ] ~integration tests~
- [ ] ~necessary TSDoc comments~

### CODE REVIEW

File: packages/experience/src/pages/Consent/OrganizationSelector/index.module.scss [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  // increase the clickable area
+  &::after {
+    content: '';
+    display: block;
+    position: absolute;
+    top: -10px;
+    left: -10px;
+    width: 40px;
+    height: 40px;
+  }
```
Comment (charIeszhao) [inline packages/experience/src/pages/Consent/OrganizationSelector/index.module.scss:27]: Why do we need to use pseudo class to increase clickable area?

File: packages/experience/src/pages/Consent/UserProfile/index.tsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+const UserProfile = ({
+  user: { id, avatar, name, primaryEmail, primaryPhone, username },
+  className,
+}: Props) => {
+  return (
+    <div className={classNames(styles.wrapper, className)}>
+      {avatar && <img src={avatar} alt="avatar" className={styles.avatar} />}
+      <div>
+        <div className={styles.name}>{name ?? id}</div>
+        <div className={styles.identifier}>{primaryEmail ?? primaryPhone ?? username}</div>
```
Comment (charIeszhao) [inline packages/experience/src/pages/Consent/UserProfile/index.tsx:20]: Is this fallback logic confirmed by designer? Especially the `name ?? id` one?

File: packages/experience/src/pages/Consent/index.tsx [hunk 3]
```diff
@@ -1,54 +1,114 @@
...
-  const { branding, color } = experienceSettings ?? {};
-  const brandingLogo = conditional(
-    branding &&
-      color &&
-      getBrandingLogoUrl({ theme, branding, isDarkModeEnabled: color.isDarkModeEnabled })
-  );
+  const { t } = useTranslation();
+
+  const [consentData, setConsentData] = useState<ConsentInfoResponse | undefined>();
+  const [selectedOrganization, setSelectedOrganization] = useState<Organization | undefined>();
```
Comment (charIeszhao) [inline packages/experience/src/pages/Consent/index.tsx:23]: nit: `| undefined` can be saved

File: packages/experience/src/pages/Consent/index.tsx [hunk 4]
```diff
@@ -1,54 +1,114 @@
...
+  }, [asyncGetConsentInfo, handleError]);
+
+  // Init the default organization selection
+  useEffect(() => {
+    if (!consentData?.organizations?.length) {
+      return;
+    }
+
+    setSelectedOrganization(consentData.organizations[0]);
+  }, [consentData]);
```
Comment (charIeszhao) [inline packages/experience/src/pages/Consent/index.tsx:64]: Seems this can be merged into the `useEffect` hook above? After `setConsentData`?

**Review by charIeszhao — APPROVED** [review]

File: packages/experience/src/pages/Consent/UserProfile/index.tsx (hunk 2 above)
Comment (simeng-li) [reply packages/experience/src/pages/Consent/UserProfile/index.tsx:20]: yes, the fallback logic is by design.  GIve the feedback to the designer, and will have some minor UI layouts updated for that later.

File: packages/experience/src/pages/Consent/index.tsx (hunk 3 above)
Comment (simeng-li) [reply packages/experience/src/pages/Consent/index.tsx]: updated

File: packages/experience/src/pages/Consent/index.tsx (hunk 4 above)
Comment (simeng-li) [reply packages/experience/src/pages/Consent/index.tsx]: updated

**Pull request merged**